addons/l10n_pk_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,436 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import re
+import json
+from odoo import models, fields, api, _
+from odoo.tools import float_is_zero
+from odoo.exceptions import RedirectWarning, UserError
+import logging
+import requests
+
+_logger = logging.getLogger(__name__)
+DEFAULT_ENDPOINT = "https://gw.fbr.gov.pk/DigitalInvoicing/v1/PostInvoiceData_v1"
+
+
+class AccountEdiFormat(models.Model):
+    _inherit = "account.edi.format"
+
+    def _is_enabled_by_default_on_journal(self, journal):
+        self.ensure_one()
+        if self.code == "pk_einvoice_1_3":
+            return False
+        return super()._is_enabled_by_default_on_journal(journal)
+
+    def _get_l10n_pk_sales_group_ids(self, company_id):
+        ChartTemplate = self.env['account.chart.template'].with_company(company_id)
+        return (
+            ChartTemplate.ref('tax_group_pk_0')
+            + ChartTemplate.ref('tax_group_pk_1')
+            + ChartTemplate.ref('tax_group_pk_2')
+            + ChartTemplate.ref('tax_group_pk_3')
+            + ChartTemplate.ref('tax_group_pk_5')
+            + ChartTemplate.ref('tax_group_pk_8')
+            + ChartTemplate.ref('tax_group_pk_10')
+            + ChartTemplate.ref('tax_group_pk_13')
+            + ChartTemplate.ref('tax_group_pk_15')
+            + ChartTemplate.ref('tax_group_pk_16')
+            + ChartTemplate.ref('tax_group_pk_17')
+            + ChartTemplate.ref('tax_group_pk_195')
+        ).ids
+
+    def _get_l10n_pk_withhold_group_ids(self, company_id):
+        return (
+            self.env['account.chart.template'].with_company(company_id).ref('tax_group_pk_wt')
+        ).ids
+
+    def _get_move_applicability(self, move):
+        # EXTENDS account_edi
+        self.ensure_one()
+        if self.code != 'pk_einvoice_1_3':
+            return super()._get_move_applicability(move)
+        sales_group_ids = self._get_l10n_pk_sales_group_ids(move.company_id)
+        is_taxable = any(line.tax_group_id.id in sales_group_ids for line in move.line_ids.tax_ids)
```

Comment:
Following the other comment on ids, you can simply do this, avoiding looping over records.

```py
    is_taxable = self._get_l10n_pk_sales_groups(move.company_id) & move.line_ids.tax_ids.tax_group_id
```

```py
In [9]: self.env['account.move'].search([]).line_ids.tax_ids.tax_group_id
Out[9]: account.tax.group(2, 1)

In [10]: (
    ...:     self.env.ref('base.be') + self.env.ref('base.it')
    ...:     & self.env.ref('base.be') + self.env.ref('base.us')
    ...: ).name
Out[10]: 'Belgium'
```